tests/draft2020-12/unevaluatedItems.json

```diff
@@ -485,5 +485,183 @@
+                }
+            ],
+            "unevaluatedItems": {
+                "multipleOf": 5
+            }
+        },
+        "tests": [
+            {
+                "description": "second item is evaluated by contains #1",
+                "data": [ 1, 2 ],
+                "valid": true
+            },
+            {
+                "description": "third item is evaluated by contains #2",
+                "data": [ 1, 2, 3 ],
+                "valid": true
+            },
+            {
+                "description": "items 3, 4 are evaluated by contains #2, #1",
+                "data": [ 1, 2, 3, 4 ],
+                "valid": true
+            },
+            {
+                "description": "item 5 not evaluated, passes unevaluatedItems",
+                "data": [ 1, 2, 3, 4, 5, 6 ],
+                "valid": true
+            },
+            {
+                "description": "item 5 not evaluated, fails unevaluatedItems",
+                "data": [ 1, 2, 3, 4, 7, 8 ],
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "unevaluatedItems and contains interact to control item dependency relationship",
+        "schema": {
+            "if": {
+                "contains": {"const": "a"}
+            },
+            "then": {
+                "if": {
+                    "contains": {"const": "b"}
+                },
+                "then": {
+                    "if": {
+                        "contains": {"const": "c"}
+                    }
+                }
+            },
+            "unevaluatedItems": false
+        },
+        "tests": [
+            {
+                "description": "empty array is valid",
+                "data": [],
+                "valid": true
+            },
+            {
+                "description": "only a's are valid",
+                "data": [ "a", "a" ],
+                "valid": true
+            },
+            {
+                "description": "a's and b's are valid",
+                "data": [ "a", "b", "a", "b", "a" ],
+                "valid": true
+            },
+            {
+                "description": "a's, b's and c's are valid",
+                "data": [ "c", "a", "c", "c", "b", "a" ],
+                "valid": true
+            },
+            {
+                "description": "only b's are invalid",
+                "data": [ "b", "b" ],
+                "valid": false
+            },
+            {
+                "description": "only c's are invalid",
+                "data": [ "c", "c" ],
+                "valid": false
+            },
+            {
+                "description": "only b's and c's are invalid",
+                "data": [ "c", "b", "c", "b", "c" ],
+                "valid": false
+            },
+            {
+                "description": "only a's and c's are invalid",
+                "data": [ "c", "a", "c", "a", "c" ],
+                "valid": false
+            }
+        ]
```

Comment:
I don't see this testing any functionality that isn't already covered. It's just an example better suited as an example in documentations somewhere.

```suggestion
```

Reply:
This test is composed of a unique set of keywords that interact in a complex way to produce a unique scenario. It's not a variant of any other test. I don't think it follows axiomatically that every arbitrary implementation that passes other tests involving subsets of these keywords will pass this test. I'd rather we include the test than make assumptions about implementations.

I think it's important that we throw a lot of different scenarios at implementations, especially for complex keywords with complex interactions. This makes it more likely that we flush out edge cases, and can only give us more confidence in the robustness of implementations.

Reply:
> I think it's important that we throw a lot of different scenarios at implementations

There are tradeoffs. Every test we have is another test we need to maintain. People have been pretty good about adding tests to the suite when they encounter bugs. I'd rather see the test suite grow that way because we know that every test has value and is worth maintaining in the long run.

That said, I do see one thing about this test that is unique. It's the only test that goes more than two levels deep. I'd be ok with keeping this test around for that reason.